Add EXPLAINER.md (#13)
guardian/dev-nginx: EXPLAINER.md

## What does dev-nginx do?

Installing and running dev-nginx will start an [nginx](https://nginx.org/en/) server instance locally on your machine.

> This instance will use any `project.conf` files found locally within the directory `/nginx/servers` to generate a virtual server host to proxy local domain url requests to the correct localhost port. You can locate this directory with the command `dev-nginx locate-nginx`.

Reviewer: ```suggestion
This instance will use any `*.conf` files found locally within the directory `/nginx/servers` to generate a virtual server host to proxy requests to localhost. You can locate this directory with the command `dev-nginx locate-nginx`.
```

> Each project config should include http directives for proxy localhost ports and necessary SSL certificates. Set these up for new projects using `dev-nginx setup-app`.

Reviewer: ```suggestion
Each project config should include http directives for proxy localhost ports and necessary SSL certificates. This is quite repetitive, so `dev-nginx` abstracts it away with the `setup-app` and `setup-cert` commands.
```

## What happens when dev-nginx is up and running locally?

1. The browser will make a request to a local development domain url, e.g. `service.local.dev-gutools.co.uk`
2. Request goes out to DNS where `*.local.dev-gutools.co.uk` is set to resolve back to `localhost` (IP address: `127.0.0.1`)

>    - An alternative to using DNS is to add a new development url entry to `/etc/hosts` file resolving to 127.0.0.1.

Reviewer: ```suggestion
   - An alternative to using DNS is to add a new development url entry to `/etc/hosts` file resolving to 127.0.0.1. This can be done with `dev-nginx add-to-hosts-file <DOMAIN>`.
```

3. Nginx server running locally receives the request

> 4. Nginx server iterates over its virtual hosts to match the request url to one of the `localhost:PORT` addresses it has from `project.conf` files, finilly proxying the request to the correct local project server instance.

Reviewer: Not quite. Nginx will search for the `server_name` that matches the request, it will then send the request to the correct local project server instance. That is, if you go to `service.local.dev-gutools.co.uk` in the browser, when nginx receives the request, it will try to work out where to send traffic by looking up the `service_name`. 

```suggestion
4. Nginx server iterates over its virtual hosts, if it finds a `server_name` that matches the request, it will then proxy the request to the correct local project server instance.
```
